platform/web/detect.py

```diff
@@ -213,6 +218,14 @@ def configure(env: "Environment"):
         env.Append(LINKFLAGS=["-s", "SIDE_MODULE=2"])
         env.extra_suffix = ".dlink" + env.extra_suffix
 
+    # Run the main application in a web worker
+    if env["do_not_block_main_thread"]:
+        env.Append(LINKFLAGS=["-s", "PROXY_TO_PTHREAD=1"])
+        env.Append(CPPDEFINES=["PROXY_TO_PTHREAD_ENABLED"])
+        env.Append(LINKFLAGS=["-s", "EXPORTED_RUNTIME_METHODS=['_emscripten_proxy_main']"])
```

Comment:
I added this exported runtime method here because otherwise, emscripten would complain that it is not exposed. It seems possible to append multiple `-s EXPORTED_RUNTIME_METHODS` parameters, as I did not have problems running the editor or a game, see my demo.